File: java/aaah.java
import java.util.Scanner;

public class aaah {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		String jon = scan.next();
		String doc = scan.next();
		if (jon.lastIndexOf('a') >= doc.lastIndexOf('a'))
			System.out.println("go");
		else
			System.out.println("no");
		scan.close();
	}
}

File: ts/src/aaah.ts
import readline = require('readline')

let lineCount = 0
let jon: string
let doc: string

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount === 0) {
      jon = line
    } else {
      doc = line
    }
    lineCount++
  })
  .on('close', () => {
    console.log(jon.lastIndexOf('a') >= doc.lastIndexOf('a') ? 'go' : 'no')
    process.exit(0)
  })
